hilda/snippets/macho/all_image_infos.py

```diff
@@ -0,0 +1,113 @@
+    'systemOrderFlags' / Int64ul,
+    'uuidArrayCount' / Int64ul,
+    '_uuidArray' / SymbolFormatField(lldb.hilda_client),
+    'uuidArray' / If(this._uuidArray != 0,
+                     Pointer(this._uuidArray, Array(this.uuidArrayCount, uuid_t))),
+    '_dyld_all_image_infos' / SymbolFormatField(lldb.hilda_client),
+    # 'dyld_all_image_infos' / Lazy(LazyBound(lambda: Pointer(this._dyld_all_image_infos, self.all_image_infos))),
+    'initialImageCount' / Int64ul,
+    'errorKind' / Int64ul,
+    '_errorClientOfDyLibPath' / SymbolFormatField(lldb.hilda_client),
+    'errorClientOfDyLibPath' / If(this._errorClientOfDyLibPath != 0, Pointer(this._errorClientofoyLibPath, CString('utf8'))),
+    '_errorTargetDylibPath' / SymbolFormatField(lldb.hilda_client),
+    'errorTargetDylibPath' / If(this._errorTargetDylibPath != 0, Pointer(this._errorTargetDylibPath, CString('utf8'))),
+    '_errorSymbol' / SymbolFormatField(lldb.hilda_client),
+    'errorSymbol' / If(this._errorSymbol != 0, Pointer(this._errorSymbol, CString('utf8'))),
+    'sharedCacheSlide' / Hex(Int64ul),
+)
+
+
+class AllImageInfos(object):
+    def reload(self):
+        with lldb.hilda_client.stopped(1):
+            all_image_infos_symbol = lldb.hilda_client.symbol(lldb.hilda_client.symbols.dyld_all_image_infos)
+            self.__all_image_infos = all_image_infos_t.parse_stream(all_image_infos_symbol)
+            self.__image_infos = []
+
+            for image_info_data in self.__all_image_infos.infoArray:
+                image_info = ImageInfo(image_info_data)
+                self.__image_infos.append(image_info)
+
+    def __init__(self):
+        self.__all_image_infos = None
+        self.__image_infos = None
+
+        self.reload()
+
+    @property
+    def images(self):
+        return self.__image_infos
+
+    def find_images(self, name):
+        return [image for image in self.images if name in image.file_path]
+
+    def image_dependencies(self, image_name):
+        images = [image for image in self.find_images(image_name)]
+
+        if len(images) == 0:
+            self.reload()
+            images = [image for image in self.find_images(image_name)]
+
+            if len(images) == 0:
+                print(f"Image {image_name} is not loaded")
+                return
+
+        return self.__image_dependencies(images)
+
+    def __image_dependencies(self, images):
+        unique_images = []
+
+        for image in images:
+            if image.file_path not in [unique_image.file_path for unique_image in unique_images]:
+                unique_images.append(image)
+
+        image = unique_images[0]
+        if len(unique_images) > 1:
+            image = self.__select_specific_image(unique_images)
+
+        dependencies = set([image_name.path for image_name in image.load_commands.dylib_commands])
+
+        return dependencies
+
+    def __select_specific_image(self, images):
+        while 1:
```

Comment:
replace this input logic with `humanfriendly`:
https://humanfriendly.readthedocs.io/en/latest/api.html#humanfriendly.prompts.prompt_for_choice

Reply:
no need for the `while` look. the `prompt_for_choice` does that for you

Reply:
`prompt_for_choice` may throw an exception though..

If that is the case, I wish to give the user another attempt to provide a valid input